java/ql/src/experimental/quantum/Analysis/WeakHashing.ql

```diff
@@ -0,0 +1,19 @@
+/**
+ * @name Weak hashes
+ * @description Finds uses of cryptographic hashing algorithms that are unapproved or otherwise weak.
+ * @id java/quantum/slices/weak-hashes
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @tags external/cwe/cwe-327
+ */
+
+import java
+import experimental.quantum.Language
+
+from Crypto::HashAlgorithmNode alg, string name, string msg
+where
+  name = alg.getAlgorithmName() and
+  not name in ["SHA256", "SHA384", "SHA512", "SHA-256", "SHA-384", "SHA-512"] and
```

Comment:
Like the comment on symmetric algorithms, the hash family and hash digest size should be used rather than string comparisons. The appropriate predicates/classes to use would be `HashType` (`getHashFamily()`) and `getDigestLength()`. 

Typing this out also makes me realize we need to standardize those predicate and class names...